pipeline_dp/pld_budget_accounting.py

```diff
@@ -0,0 +1,143 @@
+"""Privacy Loss Distribution (PLD) budget accounting for DP Pipelines."""
+
+from dataclasses import dataclass
+from pipeline_dp.aggregate_params import NoiseKind
+from dp_accounting import privacy_loss_distribution as pldlib
+
+
+@dataclass
+class MechanismSpec:
+    """Specifies the parameters for a mechanism.
+
+    NoiseKind defines the kind of distribution.
+    Sensitivity and weight are applied to the noise calculation.
+    """
+    noise_kind: NoiseKind
+    sensitivity: float = 1
+    weight: float = 1
+
+
+class PLDBudgetAccountant:
+    """Manages the privacy budget for privacy loss distributions."""
+
+    def __init__(self, total_epsilon: float, total_delta: float):
+        """Constructs a PLDBudgetAccountant.
+
+        Stores the mechanisms, the given and calculated attributes
+        of the pipeline.
+
+        Args:
+            total_epsilon: epsilon for the entire pipeline
+            total_delta: delta for the entire pipeline
+
+        Raises:
+            ValueError: Arguments are missing or out of range.
+        """
+
+        if total_epsilon <= 0:
+            raise ValueError(f"Epsilon must be positive, not {total_epsilon}.")
+
+        if total_delta < 0:
+            raise ValueError(
+                f"Delta is must be non-negative, not {total_delta}.")
+
+        self._total_epsilon = total_epsilon
+        self._total_delta = total_delta
+        self._noise = 0
+        self._mechanisms = []
+
+    def request_budget(self,
+                       noise_kind: NoiseKind,
+                       sensitivity: float = 1,
+                       weight: float = 1) -> MechanismSpec:
+        """Request a budget.
+
+        Constructs a mechanism spec based on the parameters.
+        Adds the mechanism to the pipeline for future calculation.
+
+        Args:
+            noise_kind: The kind of distribution for the mechanism.
+            sensitivity: The sensitivity for the mechanism.
+            weight: The weight for the mechanism.
+
+        Returns:
+            A mechanism spec based on the parameters.
+        """
+        mechanism_spec = MechanismSpec(noise_kind, sensitivity, weight)
+        self._mechanisms.append(mechanism_spec)
+        return mechanism_spec
+
+    def compute_budgets(self):
+        """Computes the budget for the pipeline.
+
+        Composes the mechanisms and adjusts the amount of
+        noise based on given epsilon. Sets the noise for the
+        entire pipeline.
+        """
+        if len(self._mechanisms) == 0:
+            return
+        adjusted_noise = self._compute_and_adjust()
+        self._noise = adjusted_noise
```

Comment:
This function also should update MechanismNoise objects

(more details about those object comment to request_budget above)

Here is an example in BudgetAccountant
https://github.com/OpenMined/PipelineDP/blob/65d00ee6c906a36b8a173ddbb83d7e3bf71905b4/pipeline_dp/budget_accounting.py#L106